wolfssl/wolfcrypt/asn.h

```diff
@@ -244,18 +244,19 @@ enum Misc_ASN {
     MAX_SIG_SZ          = 256,
     MAX_ALGO_SZ         =  20,
     MAX_SHORT_SZ        =   6,     /* asn int + byte len + 4 byte length */
-    MAX_SEQ_SZ          =   5,     /* enum(seq | con) + length(4) */
-    MAX_SET_SZ          =   5,     /* enum(set | con) + length(4) */
-    MAX_OCTET_STR_SZ    =   5,     /* enum(set | con) + length(4) */
-    MAX_EXP_SZ          =   5,     /* enum(contextspec|con|exp) + length(4) */
-    MAX_PRSTR_SZ        =   5,     /* enum(prstr) + length(4) */
+    MAX_SEQ_SZ          =   6,     /* enum(seq | con) + length(5) */
+    MAX_SET_SZ          =   6,     /* enum(set | con) + length(5) */
+    MAX_OCTET_STR_SZ    =   6,     /* enum(set | con) + length(5) */
+    MAX_EXP_SZ          =   6,     /* enum(contextspec|con|exp) + length(5) */
+    MAX_PRSTR_SZ        =   6,     /* enum(prstr) + length(5)     */
     MAX_VERSION_SZ      =   5,     /* enum + id + version(byte) + (header(2))*/
     MAX_ENCODED_DIG_ASN_SZ= 9,     /* enum(bit or octet) + length(4) */
```

Comment:
Does MAX_ENCODED_DIG_ASN_SZ need to increase by 1 here?

Reply:
When we look at how MAX_ENCODED_DIG_ASN_SZ is used, it is used at:
[]https://github.com/wolfSSL/wolfssl/blob/41e44fdf0744d49ed0c90ba26c0eedd3dabb3fe7/wolfcrypt/src/asn.c#L5832(url)
[https://github.com/wolfSSL/wolfssl/blob/41e44fdf0744d49ed0c90ba26c0eedd3dabb3fe7/wolfcrypt/src/asn.c#L14545](url)
https://github.com/wolfSSL/wolfssl/blob/41e44fdf0744d49ed0c90ba26c0eedd3dabb3fe7/wolfcrypt/src/asn.c#L14546[](url)

All arrays are used as argument of SetDigest(). SetDigest() doesn't call SetLenght() that we are seeing the problem. Therefore, I think we don't need to increase its length.

Reply:
MAX_DER_DIGEST_SZ and MAX_DER_DIGEST_ASN_SZ that contain MAX_ENCODED_DIG_ASN_SZ don't use SetLenght() too. As the result, I think we don't need to increase MAX_ENCODED_DIG_ASN_SZ.